internal/commands/server/k8s_e2e_test.go

```diff
@@ -123,14 +124,102 @@ func TestGatewayWithClassConfigChange(t *testing.T) {
+
+			// Create an HTTPS Gateway Listener to pass when creating gateways
+			httpsListener := createHTTPSListener(ctx, t, 443)
+
+			// Create a Gateway and wait for it to be ready
+			gatewayName := envconf.RandomName("gw", 16)
+			gateway := createGateway(ctx, t, cfg, gatewayName, gatewayClass, []gateway.Listener{httpsListener})
+
+			require.Eventually(t, func() bool {
+				err := resources.Get(ctx, gatewayName, namespace, gateway)
+				return err == nil && conditionReady(gateway.Status.Conditions)
+			}, 60*time.Second, checkInterval, "no gateway found in the allotted time")
+
+			require.Eventually(t, gatewayStatusCheck(ctx, resources, gatewayName, namespace, conditionReady), 30*time.Second, checkInterval, "no gateway found in the allotted time")
+			checkGatewayConfigAnnotation(t, gateway, gatewayClassConfig)
+
+			deployment := &appsv1.Deployment{}
+			assert.NoError(t, resources.Get(ctx, gatewayName, namespace, deployment))
+			assert.Equal(t, numberOfReplicas, *deployment.Spec.Replicas)
+
+			assert.NoError(t, resources.Delete(ctx, gateway))
+
+			return ctx
+		})
+
+	testenv.Test(t, feature.Feature())
+}
+
+func TestGatewayWithReplicasCanScale(t *testing.T) {
+	t.Parallel()
+	feature := features.New("gateway class config doesn't override manual scaling").
+		Assess("gateway deployment doesn't get overriden with kubectl scale operation", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			namespace := e2e.Namespace(ctx)
+			resources := cfg.Client().Resources(namespace)
+
+			var initialReplicas int32 = 3
+			var finalReplicas int32 = 8
+
+			// Create a GatewayClassConfig
+			gatewayClassConfig, gatewayClass := createGatewayClass(ctx, t, cfg, initialReplicas)
+
+			require.Eventually(t, gatewayClassStatusCheck(ctx, resources, gatewayClass.Name, namespace, conditionAccepted), 30*time.Second, checkInterval, "gatewayclass not accepted in the allotted time")
+
+			// Create an HTTPS Gateway Listener to pass when creating gateways
+			httpsListener := createHTTPSListener(ctx, t, 443)
+
+			// Create a Gateway and wait for it to be ready
+			gatewayName := envconf.RandomName("gw", 16)
+			gateway := createGateway(ctx, t, cfg, gatewayName, gatewayClass, []gateway.Listener{httpsListener})
+
+			require.Eventually(t, func() bool {
+				err := resources.Get(ctx, gatewayName, namespace, gateway)
+				return err == nil && conditionReady(gateway.Status.Conditions)
+			}, 60*time.Second, checkInterval, "no gateway found in the allotted time")
+
+			require.Eventually(t, gatewayStatusCheck(ctx, resources, gatewayName, namespace, conditionReady), 30*time.Second, checkInterval, "no gateway found in the allotted time")
+			checkGatewayConfigAnnotation(t, gateway, gatewayClassConfig)
+
+			deployment := &appsv1.Deployment{}
+			assert.NoError(t, resources.Get(ctx, gatewayName, namespace, deployment))
+			assert.Equal(t, initialReplicas, *deployment.Spec.Replicas)
+
+			deployment.Spec.Replicas = &finalReplicas
```

Comment:
I find the one-liner comments above to be super helpful in grepping what a test is doing. Would you mind adding one here?